### ISSUE TITLE
Memory leak with THREE.Vector3 and others

### ISSUE THREAD

**Issue by ipao9** [mrdoob/three.js#7219]
![screen shot 2015-09-22 at 1 30 45 pm](https://cloud.githubusercontent.com/assets/13321573/10030824/754a17c6-612e-11e5-907d-d97c058be7d0.png)

I'm having a memory leak from many of the THREE objects as you can see from the attached image with all the positive deltas. I am comparing a before and after of my app. I am properly calling dispose() on the geometry, texture, material. I also remove each item added to the scene from Object3D, and then remove the instance of Object3D itself. I am using this in Angular JS, latest THREE rev, chrome browser. 

I also tried setting many objects that had the "new" keyword to undefined or null to be garbage collected but found no luck. Any ideas what further work must be done to free this memory?

**Comment by WestLangley** [comment]
If you can demonstrate a three.js bug, can you provide a (simple) test case so we can reproduce your results? Please see "How to report a bug" in the [guidelines](https://github.com/mrdoob/three.js/blob/master/CONTRIBUTING.md).

**Comment by mrdoob** [comment]
For all we know, the lamp in your desk could be creating interferences with your washing machine which makes it jump at the same time the browser is trying to run the garbage collection process.

Jokes aside, try to be specific, a million things could be happening.

**Comment by ipao9** [comment]
Sorry for the generality, guys. I was hoping for maybe a quick answer for how to free THREE.Vector3. It's a little hard to replicate, especially since this is being done in Angular with several different interacting components. If possible, I'll try to explain the scenario in more depth.

I have a globe with points/markers on it. Therefore, I make a lot of "new THREE.Vector3()" calls. I tried to initialize them into variables so I can reuse them, just in case the functions that contain these calls get executed multiple times. Made no difference though. I also add various lights to my scene, and I have an Object3D instance where I add several THREE.Mesh, consisting of SphereGeometry and ShaderMaterial instantiated with "new" calls.

Below is a snippet of what my on-destroy function looks like:
https://jsfiddle.net/s5fopzy1/

Everytime I add something to the scene, I add it in sceneArr.
Everytime some geometry, mesh, or texture is newly instantiated, I added to the toDisposeArr.
However, none of the THREE.Vector3 is being handled in the destroy function even though I tried to reassign the value to null/undefined.

Let me know if more detail is needed. Much appreciated!
